src/anyp/TrafficMode.h

```diff
@@ -29,15 +24,29 @@ class TrafficMode
      */
     bool accelSurrogate = false;
 
-    /** marks ports receiving PROXY protocol traffic
+    /** marks http ports receiving PROXY protocol traffic
      *
      * Indicating the following are required:
      *  - PROXY protocol magic header
      *  - src/dst IP retrieved from magic PROXY header
      *  - indirect client IP trust verification is mandatory
      *  - TLS is not supported
      */
-    bool proxySurrogate = false;
+    bool proxySurrogateHttp = false;
+
+    // TODO: unused
+    /** marks https ports receiving PROXY protocol traffic
+     *
+     * Indicating the following are required:
+     *  - PROXY protocol magic header
+     *  - URL translation from relative to absolute form
+     *  - src/dst IP retrieved from magic PROXY header
+     *  - indirect client IP trust verification is mandatory
+     *  - Same-Origin verification is mandatory
+     *  - TLS is supported
+     *  - Squid authentication prohibited
+     */
+    bool proxySurrogateHttps = false;
```

Comment:
For now, proxySurrogateHttp and proxySurrogateHttps flags are not symmetrical because proxySurrogateHttps' usage is limited - for example, we (in PR832) do not support not intercepted HTTPS traffic including PROXY protocol header on the https_port. I outlined this also in another comment.

Reply:
> this flag only difference from proxySurrogateHttp is TLS support. That information is obtained from the tunnelSslBumping flag being set.

tunnelSslBumping is set for both "http_port ssl-bump" and "https_port ssl-bump" cases. Thus, tunnelSslBumping cannot be used to distinguish "http_port require-proxy-header" from "https_port require-proxy-header" cases.

tunnelSslBumping is also _unset_ by Squid itself in some cases, which makes its correct use for configuration-driven distinctions unreliable/tricky.

----

Overall, I see no good reason to tie, at a flag level, PROXY protocol support on https_port to SslBump. While PR 832 only adds support for SslBump cases, the flag itself reflects the actual configuration: https_port with require-proxy-header. Nothing less, nothing more. If we can keep this flag meaning this simple/direct, we should. Some implications are unavoidable or desirable, but if we can avoid an implication (without making anything worse), it is a good thing!

@eduard-bagdasaryan, does PR 832 (including this PR) contain any non-configuration code that relies on proxySurrogateHttps implying tunnelSslBumping?

Reply:
PR832 has configuration code which conditions interceptedSomewhere() on tunnelSslBumping and vice versa - so there is only an indirect proxySurrogateHttps/tunnelSslBumping dependency.

Reply:
I bet there is a misunderstanding of what proxySurrogateHttps is. This flag is not "https_port require-proxy-header". This flag is "https_port require-proxy-header ssl-bump".

Several change requests say that `surrogate traffic may be "TLS explicit proxy"`. While that is true in general, it is not true for Squid. Neither official Squid nor the PRs under review support PROXY protocol in HTTPS proxy mode! PRs under review do not set the proxySurrogateHttps flag in HTTPS proxy mode (and exclude that possibility when using the flag). 

@yadij, does the above clarify?

To resolve this misunderstanding, I suggest renaming proxySurrogateHttps to proxySurrogateHttpsSslBump and adjusting its documentation to clearly exclude ports configured for the HTTPS proxy role (i.e. https_port without ssl-bump). If somebody adds PROXY protocol support to https_port without ssl-bump, then they would split this flag in accordance to their actual use cases, without any risk that some flag use will be overlooked.

Can you think of a better way to address this misunderstanding (I have excluded one "obvious" or "nice" solution below as being much worse than the proposed ugly proxySurrogateHttpsSslBump, so please read that before responding)?

----

An alternative solution would be to let proxySurrogateHttps represent any "https_port require-proxy-header" configuration, with or without SslBump, and then use the `proxySurrogateHttps && tunnelSslBumping` condition instead of the current proxySurrogateHttps flag. That condition, if used more than once, will need to be wrapped into a method, of course (called proxySurrogateHttpsSslBump() or better).

I think this alternative should be rejected because

* tunnelSslBumping lies AFAICT -- there is Squid code that resets tunnelSslBumping from true to false in some cases. We would have to fix that code (and its implications) as well if we follow this path.
 
* we would suddenly become responsible for proper use of the new proxySurrogateHttps flag in complex contexts that Squid does not support (and may never support). That is a lot of completely unnecessary work multiplied by the likely lengthy review debates about various hypothetical use cases. Lots of time will be wasted for no immediate gain and highly questionable long-term gain (the new real use cases, if they ever surface, will most likely require changes to the otherwise unused code anyway).